Tasks/AzureCloudPowerShellDeployment/Utility.ps1

```diff
@@ -140,8 +140,25 @@ function Get-DiagnosticsExtensions($storageAccount, $extensionsPath, $storageAcc
                         $storageAccountKey = $primaryStorageKey
                     }
 
-                    Write-Host "New-AzureServiceDiagnosticsExtensionConfig -Role $role -StorageAccountName $storageAccountName -StorageAccountKey <storageKey> -DiagnosticsConfigurationPath $fullExtPath"
-                    $wadconfig = New-AzureServiceDiagnosticsExtensionConfig -Role $role -StorageAccountName $storageAccountName -StorageAccountKey $storageAccountKey -DiagnosticsConfigurationPath $fullExtPath 
+                    if((CmdletHasMember "StorageAccountName") -and (CmdletHasMember "StorageAccountKey"))
+                    {
+                        Write-Host "New-AzureServiceDiagnosticsExtensionConfig -Role $role -StorageAccountName $storageAccountName -StorageAccountKey <storageKey> -DiagnosticsConfigurationPath $fullExtPath"
+                        $wadconfig = New-AzureServiceDiagnosticsExtensionConfig -Role $role -StorageAccountName $storageAccountName -StorageAccountKey $storageAccountKey -DiagnosticsConfigurationPath $fullExtPath
+                    } 
+                    else
+                    {
+                        try
+                        {
+                            $storageContext = New-AzureStorageContext -StorageAccountName $storageAccountName -StorageAccountKey $storageAccountKey
+                            Write-Host "New-AzureServiceDiagnosticsExtensionConfig -Role $role -StorageContext $StorageContext -DiagnosticsConfigurationPath $fullExtPath"
+                            $wadconfig = New-AzureServiceDiagnosticsExtensionConfig -Role $role -StorageContext $StorageContext -DiagnosticsConfigurationPath $fullExtPath 
+                        }
+                        catch
+                        {
+                            Write-Warning "Current version of azure powershell don't support external storage account. Update version to greater than 1.3.0"
```

Comment:
This is user facing error. Can you move this to localization file. 
"Current version of azure powershell don't support external storage account for configuring diagnostics."